drivers/flash/CMakeLists.txt

```diff
@@ -20,8 +20,10 @@ zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_STM32
 endif()
 
 if(CONFIG_SOC_SERIES_STM32F3X)
-zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_STM32 flash_stm32f3x.c)
-zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_STM32 flash_stm32f3x_priv.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_STM32
+	flash_stm32.c
```

Comment:
Not an issue, just a remark for a future improvement: given that `flash_stm32.c` is now shared between all STM32 SoCs, it might be possible to compile this file as soon as `CONFIG_SOC_FLASH_STM32` is defined, and thus out of the `CONFIG_SOC_SERIES_STM32XXX` entries.